llama_cpp/llama.py

```diff
@@ -453,6 +434,7 @@ def free_lora_adapter():
         self._candidates = internals.LlamaTokenDataArray(n_vocab=self._n_vocab)
 
         self.n_tokens = 0
+        self.tokens_lora_adapters: Tuple[Tuple[str, float]] = () # Adapters that processed tokens
```

Comment:
Tracks the LoRA adapters that were used to generate the `n_tokens`. A call to `reset()` sets this to the currently active adapters.